pkg/kubectl/cmd/cp.go

```diff
@@ -125,6 +120,14 @@ func runCopy(f cmdutil.Factory, cmd *cobra.Command, out, cmderr io.Writer, args
 	if err != nil {
 		return err
 	}
+
+	if len(srcSpec.PodName) != 0 && len(destSpec.PodName) != 0 {
+		if _, err := os.Stat(args[0]); err == nil {
+			return copyToPod(f, cmd, out, cmderr, fileSpec{File: args[0]}, destSpec)
+		}
+		return cmdutil.UsageErrorf(cmd, "src doesn't exist in local filesystem")
```

Comment:
thanks, error msg reads better. If I'm understanding correctly, this patch would only allow
a user to copy a file containing a `:` to a pod, but not from a pod?

Reply:
`kubectl cp ns/pod:/a/b:c /tmp/` will succeed, `kubectl cp ns/pod:/a/b:c /tmp/dir:a/` will fail with error `src doesn't exist in local filesystem`

@juanvallejo

Reply:
As a user I would expect that if we're handling files containing `:` they would work when being copied to and from a pod.
Maybe add an `else if _, err := os.Stat(args[1]); err == nil` and copy from pod in that case, before completely failing the command?

cc @fabianofranz for an opinion here

Reply:
when we copy something from pod to local file system, the target file is unnecessary to exist first.